src/lua/datetime.lua

```diff
@@ -904,7 +903,7 @@ local function datetime_parse_from(str, obj)
     end
     check_str_or_nil(fmt, 'datetime.parse()')
 
-    local offset = 0
+    local offset
```

Comment:
Nit: I suppose we can move the declaration below since this variable is used only in the inner scope.

Reply:
I've removed sanity checks for `tzoffset` and `tz`, because these checks already present in `datetime_set` used below. Also, we don't need to call `datetime_set` if `obj` is empty.
Thanks for pointing to this!

```diff
diff --git a/src/lua/datetime.lua b/src/lua/datetime.lua
index 3ab8e5947f..0f2fcb8bfa 100644
--- a/src/lua/datetime.lua
+++ b/src/lua/datetime.lua
@@ -892,26 +892,13 @@ end
 local function datetime_parse_from(str, obj)
     check_str(str, 'datetime.parse()')
     local fmt = ''
-    local tzoffset
-    local tzname
 
     if obj ~= nil then
         check_table(obj, 'datetime.parse()')
         fmt = obj.format
-        tzoffset = obj.tzoffset
-        tzname = obj.tz
     end
     check_str_or_nil(fmt, 'datetime.parse()')
 
-    if tzoffset ~= nil then
-        local offset = get_timezone(tzoffset, 'tzoffset')
-        check_range(offset, -720, 840, 'tzoffset')
-    end
-
-    if tzname ~= nil then
-        check_str(tzname, 'datetime.parse()')
-    end
-
     local date, len
     if not fmt or fmt == '' or fmt == 'iso8601' or fmt == 'rfc3339' then
         date, len = datetime_parse_full(str)
@@ -921,7 +908,9 @@ local function datetime_parse_from(str, obj)
 
     -- Override timezone, if it was not specified in a parsed
     -- string.
-    if date.tz == '' and date.tzoffset == 0 then
+    if date.tz == '' and
+       date.tzoffset == 0 and
+       next(obj) ~= nil then
         datetime_set(date, obj or {})
     end
 
```

Reply:
I've reverted this, it's better to return error as early as possible.